graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -519,7 +509,8 @@ def _compile_ast_node_to_ir(schema, current_schema_type, ast, location, context)
     fields = _get_fields(ast)
     vertex_fields, property_fields = fields
     fragment = _get_inline_fragment(ast)
-    local_filters_directives = get_local_filter_directives(ast, vertex_fields)
+    filter_operations = get_local_filter_directives(
+        ast, current_schema_type, vertex_fields)
```

Comment:
`current_schema_type` is guaranteed to be the `root` of the context ?

Reply:
Just to clarify terminology, `context` is a mutable dict that carries all the parsing state of the frontend (like tagged values or outputs), and is not involved here.

`current_schema_type` is guaranteed to be the GraphQL type of the part of the GraphQL query to which `ast` points. So in that sense, if I understood the question correctly, the answer is yes.

Reply:
You did. Since the whole idea of correctly dealing with `@filter` directives depends on determining the scope based on where the sub-AST is rooted, I just wanted to make sure we are respecting the wishes of the previous PR.